app/helpers/fhir_helper.rb

```diff
@@ -341,4 +342,47 @@ def from_statelocal_id_extension(patient)
     statelocal_id = patient&.identifier&.find { |i| i&.system == 'http://saraalert.org/SaraAlert/state-local-id' }
     statelocal_id&.value
   end
+
+  def to_address_by_type_extension(patient, address_type)
+    case address_type
+    when 'USA'
+      FHIR::Address.new(
+        line: [patient.address_line_1, patient.address_line_2].reject(&:blank?),
+        city: patient.address_city,
+        district: patient.address_county,
+        state: patient.address_state,
+        postalCode: patient.address_zip
+      )
+    when 'Foreign'
+      [patient.foreign_address_line_1,
+       patient.foreign_address_line_2,
+       patient.foreign_address_line_3,
+       patient.foreign_address_city,
+       patient.foreign_address_country,
+       patient.foreign_address_zip,
+       patient.foreign_address_state].any? ?
```

Comment:
What's the reason that for Foreign Address we first check that it has any non-null value before creating the `FHIR::Address`, but for the USA address we create the `FHIR::Address` regardless?

Reply:
If the fields are all `nil`, then gem will ignore the `FHIR::Address` when serializing to JSON, so the `USA` address will just get ignored if all the fields are `nil`. On the `Foreign` address, we are always adding the `extension` field. If all the `Foreign` fields are `nil`, but we still add the `extension` field, that will still get serialized, so even when the Patient has no foreign addresses, they would end up with an extra blank address with just an extension on it: 
```json
        {
            "extension": [
                {
                    "url": "http://saraalert.org/StructureDefinition/address-type",
                    "valueString": "Foreign"
                }
            ]
        }
```

So to avoid this, I first check to make sure at least one `Foreign` field does exist.

Reply:
Gotcha - makes sense. Thanks!